src/transformers/models/starcoder2/modular_starcoder2.py

```diff
@@ -79,328 +78,79 @@ def forward(self, hidden_states: Optional[Tuple[torch.FloatTensor]]) -> torch.Fl
         return hidden_states
 
 
-class Starcoder2Attention(nn.Module):
-    """
-    Multi-headed attention from 'Attention Is All You Need' paper. Modified to use sliding window attention: Longformer
-    and "Generating Long Sequences with Sparse Transformers".
-    """
-
+class Starcoder2Attention(MistralAttention):
```

Comment:
init can be the same as llama as well I think we can standardize more!

Reply:
nop - the bias term in the config is different 🥲